core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -51,6 +51,31 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
+  @GET
+  @Path("executors/{executorId}/threads")
+  def threadDump(@PathParam("executorId") executorId: String): Response = withUI { ui =>
```

Comment:
You can change the return type to `Array[ThreadStackTrace]` now (and adjust the only path that actually returns something).